lms/views/basic_lti_launch.py

```diff
@@ -335,27 +310,64 @@ def _set_via_url(self, document_url):
         self.context.js_config["urls"].update(
             {"via_url": via_url(self.request, document_url)}
         )
-        self._set_submission_param("document_url", document_url)
+        self.set_canvas_submission_param("document_url", document_url)
+
+    # ---------------------------------------------------------------------- #
+    # Canvas specific functions
 
-    def _set_submission_param(self, name, value):
+    def is_launched_by_canvas(self):
+        return (
+            self.request.params.get("tool_consumer_info_product_family_code")
+            == "canvas"
+        )
+
+    def initialise_canvas_submission_params(self):
+        """
+        Add config used by frontend to call Canvas `record_submission` API.
+
+        The outcome reporting params are typically only available when
+        students (not teachers) launch an assignment.
+        """
+
+        lis_result_sourcedid = self.request.params.get("lis_result_sourcedid")
+        lis_outcome_service_url = self.request.params.get("lis_outcome_service_url")
+
+        if lis_result_sourcedid and lis_outcome_service_url:
+            self.context.js_config["submissionParams"] = {
+                "h_username": self.context.h_user.username,
+                "lis_result_sourcedid": lis_result_sourcedid,
+                "lis_outcome_service_url": lis_outcome_service_url,
+            }
+
+    def set_canvas_submission_param(self, name, value):
         """Update config for frontend's calls to `report_submisssion` API."""
 
         if "submissionParams" in self.context.js_config:
             self.context.js_config["submissionParams"][name] = value
 
-    def _set_focused_user(self, username):
+    def set_canvas_focused_user(self):
         """Configure the Hypothesis client to focus on a particular user."""
 
+        # If the launch has been configured to focus on the annotations from
+        # a particular user, translate that into Hypothesis client configuration.
+
+        # This parameter is only passed as a part of Canvas Speedgrader config
+        # and is passed as a parameter to a URL which they call us back on.
+        focused_user = self.request.params.get("focused_user")
+        if not focused_user:
+            return
+
         h_api = self.request.find_service(name="h_api")
 
         try:
-            display_name = h_api.get_user(username).display_name
+            display_name = h_api.get_user(focused_user).display_name
         except HAPIError:
             # If we couldn't fetch the student's name for any reason, fall back
             # to a placeholder rather than giving up entirely, since the rest
             # of the experience can still work.
             display_name = "(Couldn't fetch student name)"
 
+        # TODO! - Could/should this be replaced with a LISSourcedId lookup?
```

Comment:
```suggestion
        # TODO! - Could/should this be replaced with a LISResultSourcedId lookup?
```

I think the problem, unfortunately, is that we don't save `LISResultSourcedId`'s to the DB for Canvas. We could start doing so but we'd have to maintain the API lookup as a fallback for preexisting assignments. Other than that it feels like we probably should do it